agent/node.go

```diff
@@ -633,7 +634,23 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			ready = nil
 		}
 
-		err = n.waitRole(ctx, ca.AgentRole)
+		roleChanged := make(chan struct{})
+		go func() {
+			n.waitRole(ctx, ca.AgentRole)
```

Comment:
Right. If the m.Run returns not because of demotion or context canceled, the goroutine would leak. I will fix it.